packages/keyring-controller/src/KeyringController.ts

```diff
@@ -364,6 +364,15 @@ export type ExportableKeyEncryptor = GenericEncryptor & {
   importKey: (key: string) => Promise<unknown>;
 };
 
+export type KeyringSelector =
+  | {
+      type: string;
+      index?: number;
+    }
+  | {
+      address: Hex;
+    };
```

Comment:
This comment is beyond the scope of the PR

How should we ensure the correct execution of an operation when we have multiple keyrings with the same address/account? @Gudahtt, and I believe others, have already pointed this particular issue regarding the `removeAccount` method

Reply:
Currently every time a new keyring is added (or deserialized from vault) `this.#checkForDuplicates` is called, but not when adding a new account on a keyring.

This means that a user would be able to add an account that is already on another keyring without issues, but it would then get errors when unlocking the wallet, because the keyrings would be added along with all the accounts already, and `checkForDuplicates` in `#newKeyring` would throw. 

Perhaps we could check for duplicates also when adding new accounts?

Reply:
>Perhaps we could check for duplicates also when adding new accounts?

This sounds like an improvement for now, for as long as we're using the address as the unique identifier. Eventually we'll be moving away from that idea hopefully

Reply:
> Perhaps we could check for duplicates also when adding new accounts? it would solve the root issue also regarding removeAccount as well

Hmm, I'm thinking that this could lead to interesting scenarios as well: if a user imports an account which can also be derivable with e.g. `index + 1` of the HD keyring, it would end up in a situation where no more HD accounts can be added until the imported one is removed, and it would also receive no clear error or warning about that.

Same applies for Snaps: I'm wondering if a Snap could potentially block account additions on other keyrings

Reply:
> for as long as we're using the address as the unique identifier. Eventually we'll be moving away from that idea hopefully

The `accounts-controller` is introducing the concept of account ID (an UUID) in the clients, it would be good to also use this identifier inside the `keyring-controller`

> I'm wondering if a Snap could potentially block account additions on other keyrings, or the other way around

Yes, this is a possible scenario, I think the long term solution is to use the regular IDs instead of addresses as the identifier. It's a good UX to be able to sign with different keyrings if the user wants to so I don't think we should block the account duplication at a `keyring-controller` but simplify it at a UI/UX level